packages/story-editor/src/components/checklist/checks/pageBackgroundLowTextContrast/component.js

```diff
@@ -45,30 +45,14 @@ const PageBackgroundTextLowContrast = () => {
     height: pageHeight,
   }));
 
-  const getFailingPages = useCallback(async () => {
-    const promises = [];
-    (storyPages || []).forEach((page) => {
-      const maybeTextContrastResult = pageBackgroundTextLowContrast({
-        ...page,
-        pageSize,
-      });
-      if (maybeTextContrastResult instanceof Promise) {
-        promises.push(
-          maybeTextContrastResult.then((result) => ({ result, page }))
-        );
-      } else {
-        promises.push(maybeTextContrastResult);
-      }
-    });
-    const awaitedResult = await Promise.all(promises);
-    return awaitedResult.filter(({ result }) => result).map(({ page }) => page);
-  }, [storyPages, pageSize]);
-
   useEffect(() => {
-    getFailingPages().then((failures) => {
-      setFailingPages(failures);
-    });
-  }, [getFailingPages]);
+    getPagesWithFailedContrast(storyPages, pageSize)
+      .then((failures) => {
+        const failedPages = failures.map(({ page }) => page);
+        failedPages.length && setFailingPages(failedPages);
```

Comment:
Ahh I think this may be a bug.

Lets run through the scenario:
- a story page has a white background, a black square, and some black text, with the black text positioned over the background
- you move the text to be on top of the black square
- this triggers a failing page
- you open the help center and see this check
- with the help center open, you move the text to be positioned back over the background
- this check should go away, but doesn't because it only sets state if `failedPages.length`

Reply:
Also small (and probably very rare) race case to consider is if a user is able to update the story multiple times before this check completes.

We may want some sort of stale flag to ensure that we're always deriving failed pages off of the freshest version of `storyPages`. Something like:
```js
useEffect(() => {
  // can name this whatever you want
  let isInputStale = false;
  getPagesWithFailedContrast(storyPages, pageSize)
      .then((failures) => {
        if (isInputStale) {
          return;
        }
        const failedPages = failures.map(({ page }) => page);
        setFailingPages(failedPages);
      })
      .catch(() => {
        // maybe add a TODO comment in here or something :shrug:
      });
     
  return () => {
    isInputStale = true;
  }
}, [storyPages, pageSize]);
```

Reply:
Great catch. Not sure how I overlooked that.